src/tty-ask-password-agent/tty-ask-password-agent.c

```diff
@@ -313,6 +325,39 @@ static int wall_tty_block(void) {
         return fd;
 }
 
+static int parse_query(const char *arg) {
+        const char *l;
+        bool strict = false;
+        int n, r;
+
+        assert(arg || !strict);
+
+        if (!arg)
+                goto default_noarg;
+
+        if (streq(arg, "all")) {
+                arg_queryPID = 0;
+                return 1;
+        }
+
+        l = startswith(arg, "pid=");
+
+        r = safe_atoi(l /* "--query=pid=XXX" */ ?: arg /* "--query=XXX" */, &n);
```

Comment:
we have parse_pid() btw.

Reply:
not sure about the usage of this one, tried something :+1: